cinema/views.py

```diff
@@ -124,6 +161,14 @@
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed for this resource")
```

Comment:
Raising a `NotFound` exception for unsupported actions is misleading. Consider using `MethodNotAllowed` to indicate that the HTTP method is not allowed for the requested resource.